File: api/Models/HubPictureSuggestion.cs
using System.Text.Json.Serialization;

namespace CroApp.Api.Models;

// One row per photo suggested for a Hub, awaiting admin review - mirrors the Hub
// location suggestion/approval flow (see HubStatus.Pending), but doesn't need its own
// status partition since rejecting a suggestion just deletes the row outright, same as
// rejecting a pending Hub. HubId is the partition key: the dominant per-hub query is
// "pending photo suggestions for this hub", same /hubId reasoning as HubMessage. The
// admin moderation feed (every pending suggestion across every Hub) is a small-volume
// cross-partition scan, acceptable at a curated Hub list's size.
public record HubPictureSuggestion(
    [property: JsonPropertyName("id")] string Id,
    string HubId,
    string SuggestedByUserId,
    string BlobUrl,
    DateTimeOffset CreatedAt);

File: app/lib/models/hub_picture_suggestion.dart
// A user-suggested photo for a Hub, pending admin approval - mirrors
// api/Models/HubPictureSuggestion.cs. The suggester's username is resolved separately
// (not carried on this model) the same way hub location suggestions resolve theirs.
class HubPictureSuggestion {
  final String id;
  final String hubId;
  final String suggestedByUserId;
  final String blobUrl;
  final DateTime createdAt;

  HubPictureSuggestion({
    required this.id,
    required this.hubId,
    required this.suggestedByUserId,
    required this.blobUrl,
    required this.createdAt,
  });

  factory HubPictureSuggestion.fromJson(Map<String, dynamic> json) => HubPictureSuggestion(
        id: json['id'] as String,
        hubId: json['hubId'] as String,
        suggestedByUserId: json['suggestedByUserId'] as String,
        blobUrl: json['blobUrl'] as String,
        createdAt: DateTime.parse(json['createdAt'] as String),
      );
}
